assets/tutorials/multiple_feasible_solutions.jl

```diff
@@ -0,0 +1,240 @@
+# We will also see how to enforce an "all-different" constraint on a set of integer variables.
+
+# We start with some objects from recreational mathematics that could be called **"symmetric square sums"**.
+# Here are a few examples (inspired by [Number Squares](https://www.futilitycloset.com/2012/12/05/number-squares/)):
+"""
+    1529      2318      5219
+    5837      3790      2384
+    2340      1956      1867
+  + ____    + ____    + ____
+    9706      8064      9470
+"""
+
+# Notice how all the digits 0 to 9 are used at least once,
+# the first three rows sum to the last row,
+# and the columns in each are the same as the corresponding rows (forming a symmetric matrix).
+
+# We might ask: how many such squares are there?
+# In this tutorial, we will find out the answer
+# and see how to go about answering this and similar questions.
+
+# This tutorial uses the following packages:
+using JuMP
+import Gurobi
+
+# Here we are using [Gurobi](https://github.com/jump-dev/Gurobi.jl) 
+# since it provides the required functionality
+# for this example (i.e. finding multiple feasible solutions).
+# Gurobi is a commercial solver, that is, a paid license is needed
+# for those using the solver for commercial purposes. However there are
+# trial and/or free licenses available for academic and student users.
+
+# ## Model Specifics
+
+solver = Gurobi.Optimizer
+model = JuMP.Model(solver)
+
+# We need to set specific Gurobi parameters to enable the
+# [multiple solution functionality](https://www.gurobi.com/documentation/9.0/refman/finding_multiple_solutions.html).
+
+# The first setting turns on the exhaustive search mode for multiple solutions:
+JuMP.set_optimizer_attribute(model, "PoolSearchMode", 2)
+
+# The second sets a limit for the number of solutions found. 
+# Here the value 100 is an arbitrary but large enough whole number
+# for our particular model (but will depend on the application in general).
+JuMP.set_optimizer_attribute(model, "PoolSolutions", 100)
+
+## %%  #src
+# ## Setting up the model:
+
+# We are going to use 4-digit numbers:
+number_of_digits = 4
+
+# Let's define the index sets for our variables and constraints.
+# We keep track of each "place" (units, tens, one-hundreds, one-thousands):
+PLACES = 0:(number_of_digits-1)
+# The number of rows of the symmetric square sums are the same as the number of digits:
+ROWS = 1:number_of_digits
+
+# Next, we define the model's core variables.
+# Here a given digit between 0 and 9 is found in the `i`-th row at the `j`-th place:
+@variable(model, 0 <= Digit[i = ROWS, j = PLACES] <= 9, Int)
+# We also need a higher level "term" variable that represents the actual number in each row:
+@variable(model, Term[ROWS] >= 1, Int)
+# The lower bound of 1 is because we want to get back non-zero solutions.
+
+# Defining the objective function,
+# since this is a feasibility problem, it can just be set to 0:
+@objective(model, Max, 0)
```

Comment:
I think it's preferable to just not set an objective.

Reply:
Ok. Changed.